mrpt_graphslam_2d/CMakeLists.txt

```diff
@@ -18,3 +18,47 @@ if (BUILD_TESTING)
     ${test_output_directory}/${PROJECT_TEST_NAME})
 endif()
 
+set(PROJECT_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../)
+
+#V-REP Config
+################################################################################
+
+if (EXISTS "$ENV{VREP}")
+  message(STATUS "Found V-REP")
+  set (V-REP_ROOT "${PROJECT_ROOT_DIR}/../" CACHE "V-REP root dir" STRING)
+
+
+  set (V-REP_PATH "${PROJECT_ROOT_DIR}/../V-REP/programming/" CACHE "The V-REP build path" STRING)
+
+  set (SOURCES remoteApi/bubbleRob_test.cpp
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApi.c
+    ${V-REP_ROOT}/V-REP/programming/common/shared_memory.c
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApi.h
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApiPlatform.h
+    ${V-REP_ROOT}/V-REP/programming/include/shared_memory.h
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApiPlatform.c)
+
+  set (EXTRA_LIBRARIES pthread)
+
+  include_directories (${V-REP_PATH}/remoteApi ${V-REP_PATH}/include)
+  
+  add_definitions( -Dpthread )
+  add_definitions (-DNON_MATLAB_PARSING)
+  add_definitions (-DMAX_EXT_API_CONNECTIONS=255)
+  
+  add_library (test_Client SHARED ${SOURCES})
+  
+  
+  add_executable(bubbleRob_test ${SOURCES} )
+
+  target_link_libraries (bubbleRob_test test_Client ${EXTRA_LIBRARIES})
+  target_link_libraries (bubbleRob_test test_Client rt)
```

Comment:
You don't have a consistent way of setting the libraries you link against.

Either add all of them to a CMake list and then link against that or list them one by one in the `target_link_libraries` call.

target_link_libraries(
    bubbleRob_test
    test_Client
    pthread
    rt)